homeassistant/helpers/intent.py

```diff
@@ -57,7 +57,8 @@
 def async_register(hass: HomeAssistant, handler: IntentHandler) -> None:
     """Register an intent with Home Assistant."""
     if (intents := hass.data.get(DATA_KEY)) is None:
-        intents = hass.data[DATA_KEY] = {}
+        intents = {}
+        hass.data[DATA_KEY] = intents
```

Comment:
mypy made me change this even though I didn't modify it :man_shrugging:

Reply:
Replace this with `dict.setdefault():`
```py
intents = hass.data.setdefault(DATA_KEY, {})
```

Reply:
I tried that, and it complains about it as well: `error: Cannot infer type argument 1 of "setdefault" of "HassDict"`